src/main/java/com/selfxdsd/storage/SelfContracts.java

```diff
@@ -132,35 +133,45 @@ public Contract addContract(
                 + " project. The Project with " + repoFullName
                 + " and " + provider + " was not found.");
         }
-        final Contributor contributor = this.storage.contributors()
-            .getById(contributorUsername, provider);
-        if (contributor == null) {
-            throw new IllegalStateException("Can't attach the Contract to"
-                + " contributor. The Contributor with "
-                + contributorUsername + " and " + provider
-                + " was not found.");
-        }
-        final InsertOnDuplicateStep<?> insert = jooq
-            .insertInto(SLF_CONTRACTS_XDSD,
-                SLF_CONTRACTS_XDSD.REPO_FULLNAME,
-                SLF_CONTRACTS_XDSD.USERNAME,
-                SLF_CONTRACTS_XDSD.PROVIDER,
-                SLF_CONTRACTS_XDSD.HOURLY_RATE,
-                SLF_CONTRACTS_XDSD.ROLE)
-            .values(repoFullName, contributorUsername,
-                provider, hourlyRate.longValueExact(), role);
-        final int execute;
-        if (database.dbms().equals(Database.Dbms.MY_SQL)) {
-            execute = insert.onDuplicateKeyIgnore().execute();
-        } else {
-            execute = insert.execute();
-        }
-        if (execute != 1) {
+        final List<Object> txResults;
+        try {
+            txResults = this.transactionResultCompat(jooq, conf -> {
```

Comment:
@criske Is this strange method really needed? I used simply ``jooq.transaction(...)`` [here](https://github.com/self-xdsd/self-storage/blob/550138abb986b3f3aa9279de24bbdd62712f4605/src/main/java/com/selfxdsd/storage/SelfWallets.java#L147).

Reply:
@amihaiemil I could have sworn that last time when I've tried transactions they didn't work (I think was in `SelfProjects`).
So for this I didn't bother to check again.
I'll revert back to `jooq.transaction(...)` :D